packages/server-wallet/src/protocols/state.ts

```diff
@@ -50,47 +49,54 @@ type SignedByMe = {latestSignedByMe: SignedStateWithHash};
 export type ChannelStateWithMe = ChannelState & SignedByMe;
 export type ChannelStateWithSupported = ChannelState & SignedByMe & WithSupported;
 
-export type Stage = 'Missing' | 'PrefundSetup' | 'PostfundSetup' | 'Running' | 'Final';
 /**
- *
- * @param state: SignedStateWithHash | undefined
- * Returns stage of state.
- *
- * Useful for partitioning the protocol state to decide what action to next take.
+ * The definition of a ChannelStatus is still under debate, but for now
+ * this is what we have settled on. Future changes should include 'funding',
+ * 'defunding', and state related to challenging, probably. See this issue
+ * for more: https://github.com/statechannels/statechannels/issues/2509
  */
-export const stage = (state: State | undefined): Stage =>
-  !state
-    ? 'Missing'
-    : state.isFinal
-    ? 'Final'
-    : state.turnNum <= state.participants.length - 1
-    ? 'PrefundSetup'
-    : state.turnNum <= state.participants.length * 2 - 1
-    ? 'PostfundSetup'
-    : 'Running';
+export const status = (channelState: ChannelState): ChannelStatus => {
+  const {supported, latest, latestSignedByMe, support} = channelState;
+  const {participants} = supported ?? latest;
+  if (latest.isFinal) {
+    if (support?.every(s => s.isFinal)) {
+      return 'closed'; // the entire support chain isFinal
+    } else {
+      return 'closing'; // at least one isFinal state proposed
+    }
+  } else {
+    if (latest.turnNum >= participants.length * 2) {
```

Comment:
In this branch, we know that `latest.isFinal` is true. That does not necessarily imply that there are no final states.

Reply:
I think it does (ref: https://github.com/statechannels/statechannels/pull/2857#discussion_r523023180)

Reply:
Maybe I am confused because I don't know what branch I am in. I think this function is difficult to understand.

Reply:
Why can't it be the case that we have a supported `isFinal: true` state, but Bob sent a `isFinal: false` state with a higher turn number?